internal/dataplane/parser/parser.go

```diff
@@ -461,13 +458,8 @@ func getGatewayCerts(log logrus.FieldLogger, s store.Storer) []certWrapper {
 						}, grants)
 
 						if !newRefChecker(ref).IsRefAllowedByGrant(allowed) {
-							log.WithFields(logrus.Fields{
-								"gateway":           gateway.Name,
-								"gateway_namespace": gateway.Namespace,
-								"listener":          listener.Name,
-								"secret_name":       string(ref.Name),
-								"secret_namespace":  namespace,
-							}).WithError(err).Error("secret reference not allowed by ReferenceGrant")
+							secretName := namespace + "/" + string(ref.Name)
+							p.registerTranslationFailure(fmt.Sprintf("secret reference (%s) not allowed by ReferenceGrant", secretName), gateway)
```

Comment:
Ditto, handled on the controller side:
https://github.com/Kong/kubernetes-ingress-controller/blob/ec0f2702b99f8a75581e5c789bdf57a062ad1491/internal/controllers/gateway/gateway_utils.go#L442